plugin.php

```diff
@@ -97,14 +150,27 @@ function show_save_admin_notice() {
  *
  * @return string[]
  */
-function get_site_urls( $config = null ) {
+function get_site_urls( $config = null, $page = 1, $posts_per_page = -1 ) {
```

Comment:
As per our discussion on slack, I feel like moving pagination out of the function will not be a good idea by considering the filter `static_page_site_urls` and its usage.

It will be good to not change the whole logic here,
Instead of 
`$posts = get_posts( [ 'posts_per_page' => -1, 'post_type' => 'any' ] );`
use WP_Query  with out without paggination ( Fine with both - considering we are only fetching IDs )
```
$query_args = [
		'post_type'              => 'any',
		'posts_per_page'         => -1,
		'fields'                 => 'ids',
		'update_post_meta_cache' => false,
		'update_post_term_cache' => false,
	];
```
And use `array_chunk` in outer function to split the array in chunk 50. and schedule event in loop.